### PR TITLE
[PT FE]: fix constant folding for dequantization

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *added transformation for tuple construct + tuple unpack pattern, occurs in non-frozen models*
 - *added mark dequantize subgraph transformation on frontend side*

### Tickets:
 - *CVS-113593*

### CODE REVIEW

File: src/frontends/pytorch/src/frontend.cpp [hunk 1]
```diff
@@ -155,6 +157,7 @@ void FrontEnd::normalize(const std::shared_ptr<ov::Model>& model) const {
     manager.register_pass<ov::pass::ConvertConvertLike>();
     manager.register_pass<ov::frontend::pytorch::pass::AtenIndexToSelect>();
 
+    manager.register_pass<ov::pass::MarkDequantizationSubgraph>(element::TypeVector{element::u8});
```
Comment (slyalin) [inline src/frontends/pytorch/src/frontend.cpp:160]: Why only u8? List all types that we can support in theory (u8, i8, i4, u4).

File: src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        auto tuple_construct = cast_fw_node(input_node, "prim::TupleConstruct");
+        if (!tuple_construct) {
+            return false;
+        }
+        for (const auto& input : input_node->inputs()) {
+            const auto& out = input.get_source_output();
+            outputs.push_back(out);
+            to_copy_rt.push_back(out.get_node_shared_ptr());
+        }
+        copy_runtime_info({tuple_unpack, input_node}, to_copy_rt);
```
Comment (slyalin) [inline src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp:36]: Not sure that this is a correct RT transfer. `to_copy_rt` contains all nodes that provides inputs to `TupleConstruct`, why do we need to fuse RT into all of them? Looks like there is no any other place suitable for that but can we just lose this info? What are the implications?

File: src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+PrimTupleUnpackReplacer::PrimTupleUnpackReplacer() {
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp:17]: added

File: src/frontends/pytorch/src/frontend.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/frontend.cpp]: added more precisions

File: src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/transforms/tuple_unpack_replacer.cpp]: I think there is no more impack here if we will no copy info as there is no additional context between packing and unpacking of these nodes, so I removed copying

**Review by slyalin — APPROVED** [review]

**Pull request merged**